stack.yaml

```diff
@@ -131,3 +142,11 @@ ghc-options:
   cardano-sl:            -Werror
   cardano-sl-tools:      -Werror
   cardano-sl-lwallet:    -Werror
+  cardano-sl-explorer:   -Werror
+
+# Added for explorer
+flags:
+  cardano-sl:
+    with-explorer: True
```

Comment:
1. Yes, that's the current issue.
2. The things is, there is no point of passing `with-explorer` since we need it in both cases - it's due to gold linker issues. As you wrote yourself, we must enable `with-explorer` for `cardano-sl`.
3. I can do it, but it's connected to the next point.
4. Ok, I can fix this, but I want to be sure we understand each other. We have three code chunks that we can group for _explorer_.
a. the base `cardano-sl` code.
b. the base `cardano-sl` code related to `explorer`, which we enable/disable using the stupid flag.
c. the actual `explorer` code. This is being built on its own since it's in `stack.yaml` `packages`

The issues seem to come from b. but we are actually already building _explorer_ due to c. So, what I can see as a solution for 3. is to remove the flag entirely.

We can't build this project with `explorer` inside it if we pass `False` as `with-explorer`. If I return the flag `with-explorer` to _False_.
When I don't pass a flag:
```    /home/ksaric/.nix-profile/bin/ld.gold: --hash-size=31: unknown option
    /home/ksaric/.nix-profile/bin/ld.gold: use the --help option for usage information
    collect2: error: ld returned 1 exit status
    `gcc' failed in phase `Linker'. (Exit code: 1)
```
When I pass a flag with `stack build --fast -j4 --flag cardano-sl:with-explorer`, all good.